src/styleguide/components/s-api-mock-test.vue

```diff
@@ -0,0 +1,379 @@
+
+  interface IData {
+    search: string;
+    handlers: ReadonlyArray<RequestHandler> | null;
+    configurations: {
+      [key: string]: IDebugConfiguration;
+    },
+  }
+
+  /**
+   * Lists all active Mock API Endpoint handlers and allows enabling error mode for them.
+   */
+  export default defineComponent({
+    name: 's-api-mock-test',
+    components: { eSelect, eCheckbox },
+
+    // props: {},
+    emits: {
+      close: null,
+    },
+
+    setup(): ISetup {
+      return {
+        storageKey: 'sApiMockTest',
+        statusOptions: [
+          {
+            label: '200 OK',
+            value: '200',
+          },
+          {
+            label: '400 Bad Request',
+            value: '400',
+          },
+          {
+            label: '403 Forbidden',
+            value: '403',
+          },
+          {
+            label: '500 Internal Server Error',
+            value: '500',
+          },
+        ],
+      };
+    },
+    data(): IData {
+      return {
+        search: '',
+        handlers: null,
+        configurations: {},
+      };
+    },
+
+    computed: {
+      /**
+       * Returns list of endpoints  (filtered by search).
+       */
+      filteredEndpoints(): object[] | null {
+        const { search, handlers } = this;
+
+        if (!handlers) {
+          return null;
+        }
+
+        const mappedHandlers = handlers.map(handler => handler.info);
+
+        if (!search) {
+          return mappedHandlers;
+        }
+
+        return mappedHandlers.filter(endpoint => endpoint.header.toLowerCase().includes(search.toLowerCase()));
+      },
+    },
+    watch: {
+      configurations: {
+        /**
+         * Updates handler and local storage when configuration changed.
+         */
+        handler(configurationsValue: IDebugConfiguration): void {
+          this.setupHandlers();
+          window.localStorage.setItem(this.storageKey, JSON.stringify(configurationsValue));
+        },
+        deep: true,
+      },
+    },
+
+    // beforeCreate() {},
+    created() {
+      mockWorker.resetHandlers();
+
+      this.handlers = mockWorker.listHandlers();
+      this.setDefaultConfigurations();
+
+      const configurationsFromStorage = JSON.parse(window.localStorage.getItem(this.storageKey) || '{}');
+
+      if (configurationsFromStorage) {
+        this.configurations = {
+          ...this.configurations,
+          ...configurationsFromStorage,
+        };
+
+        this.setupHandlers();
+      }
+    },
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    beforeUnmount() {
+      mockWorker.resetHandlers();
+    },
+    // unmounted() {},
+
+    methods: {
+      /**
+       * Sets up the default configurations.
+       */
+      setDefaultConfigurations(): void {
+        this.handlers?.forEach((handler) => {
+          const { header } = handler.info;
+
+          this.configurations[header] = {
```

Comment:
suggestion: not sure if this is still a thing with Vue 3 but in Vue 2 we would have to use `Vue.set` for such a manipulation.

Reply:
That is not needed anymore in Vue because it uses proxy detection.

https://vuejsdevelopers.com/2017/03/05/vue-js-reactivity/